src/cuepoints.ts

```diff
@@ -0,0 +1,38 @@
+import {CuepointService} from './cuepoint-service';
+import Player = KalturaPlayerTypes.Player;
+
+export class Cuepoints extends KalturaPlayer.core.BasePlugin {
+  private _cuePointService: CuepointService;
+  /**
+   * The default configuration of the plugin.
+   * @static
+   */
+  static defaultConfig = {};
+
+  constructor(name: string, player: Player) {
+    super(name, player);
+    this.eventManager.listen(this.player, this.player.Event.CHANGE_SOURCE_ENDED, () => {
+      this._cuePointService.init();
+    });
+    this._cuePointService = new CuepointService(player, this.logger);
+    player.registerService('cuepoints', this._cuePointService);
```

Comment:
how can we make sure another dependent plugin will only be initialised after service was added?
Three options -
1. expose services from plugins with special CTOR (like middleware etc) and only when all plugins are loaded emit an event that other plugins can depend on.
2. On each service registration emit an event from player and plugin that needs it can do a check if it exists and if not register to the event and check if the event is of the service it needs - don't like this too much.
3. make services dependencies of plugin and use the async plugin init to wait for it. Player will flag to plugin when service is ready, and if all plugins are initialized and no service is available then fail/reject it to plugin so plugin can finish init and do error handling.